app/uk/gov/hmrc/fileupload/controllers/EnvelopeController.scala

```diff
@@ -27,27 +29,41 @@ import uk.gov.hmrc.fileupload.utils.JsonUtils.jsonBodyParser
 import uk.gov.hmrc.fileupload.write.envelope._
 import uk.gov.hmrc.fileupload.write.infrastructure._
 import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, _}
-import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
 
+
+package object envelope {
+
+  import play.api.libs.concurrent.Execution.Implicits._
+
+  object EnvelopeController extends EnvelopeController(
+    withBasicAuth = MicroserviceGlobal.withBasicAuth,
```

Comment:
I think we should do it the play 2.5 way:
http://www.schibsted.pl/blog/dependency-injection-play-framework-scala/#manual

Code example:
https://github.com/povder/play-di/tree/master/manual

Play doc:
https://www.playframework.com/documentation/2.5.x/ScalaCompileTimeDependencyInjection